packages/tradershub/src/components/Dialog/DialogHeader.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Heading, qtMerge } from '@deriv/quill-design';
+import CloseIcon from '../../public/images/ic-close-dark.svg';
+import { useModal } from '../ModalProvider';
+
+/**
+ * Type for the DialogHeader component props
+ * @typedef TDialogHeader
+ * @property {string} [className] - Optional CSS class name
+ * @property {boolean} [hideCloseButton] - Optional flag to hide the close button
+ * @property {string} [title] - Optional title for the dialog header
+ */
+type TDialogHeader = {
+    className?: string;
+    hideCloseButton?: boolean;
```

Comment:
If it is boolean value then name should be `hasCloseButton` 
It is to follow one naming convention 
for flag it should be action(hasSomething e.g. hasCloseButton)
hideCloseButton sounds like an action(function) meanwhile it is a boolean value here

Reply:
I think the naming conventions is brought over from `wallets` where the reasoning behind, is so when you passed the props we checked for negative cases, for example:

in the case of `hasCloseButton` is `true` by default
 `<Dialog.Header />` but if you want to set it to false it will be as this `<Dialog.Header hasCloseButton={false} />`

It will be like this:
in the case of `hideCloseButton` is `false` by default
`<Dialog.Header />` but if you want to set it to false it will be as this `<Dialog.Header hideCloseButton />`

Not much of difference in terms of logic but they way we are writing it as a user of the component I believe it's much more intuitive and more straight forward and you don't have to passed any `boolean` values in some cases for example like the one above.

Let me know if you have any objections over this approach I believe this should be agreed as a team so we all can be on the same page 😃 👍🏼 

cc: @mahdiyeh-deriv @thisyahlen-deriv @hirad-deriv @shontzu-deriv

Reply:
I agree with @aizad-deriv

Reply:
And also to add the above is that the default behavior of a dialog it has a close button so having a prop that says `hasCloseIcon` seems pointless hence why having a prop to hide it in my opinion will be more of use.

Reply:
I felt like the second case, **`hideCloseButton` is `false` by default**, is kind of confusing even though it's just the opposite of the first case

Reply:
My concerns was regarding the naming convention and action. Just checked with Material UI and they are using the same so LGTM

Reply:
@shontzu-deriv wait there I'll explain to you 😂